src/backend-api/Controllers/ContextController.cs

```diff
@@ -29,5 +29,30 @@ public string VerbatimMatch(string routeModel)
             return routeModel.ToLower() == "laptop" ? "computer" : routeModel.ToLower();
         }
 
+        public void UpdateHardwareHistory(bool isAssigned, int employeeId, int hardwareId, string type)
+        {
+            // Update the history: Assigned or Unassigned
+            _context.HardwareHistory.Add(new HardwareHistory
+            {
+                HardwareId = hardwareId,
+                EmployeeId = employeeId,
+                HardwareType = type,
+                EventType = $"{(isAssigned ? "Assigned" : "Unassigned")}",
+                EventDate = DateTime.Now,
+            });
+        }
+
+        public ProgramHistory UpdateProgramHistory(bool isAssigned, int employeeId, int programId)
+        {
+            // Update the history: Assigned or Unassigned
+            return (new ProgramHistory
```

Comment:
Is there a reason as to why the `UpdateHardwareHistory` is a void method and `UpdateProgramHistory` returns a `ProgramHistory` object?

Reply:
Because program histories can be stacked as to save the database (using addrange) but it didnt make sense to me to do that with the hardware updates